github/table_github_team.go

```diff
@@ -0,0 +1,205 @@
+		// {Name: "updated_at", Type: pb.ColumnType_TIMESTAMP, Hydrate: tableGitHubTeamGet},
+		{Name: "url", Type: pb.ColumnType_STRING, Description: "The API URL of the team."},
+	}
+}
+
+func gitHubOrgTeamColumns() []*plugin.Column {
+	teamColumns := gitHubTeamColumns()
+	orgColumns := []*plugin.Column{
+		{Name: "organization", Type: pb.ColumnType_STRING, Description: "The organization the team is associated with.", Transform: transform.FromQual("organization")},
+	}
+
+	return append(teamColumns, orgColumns...)
+}
+
+func tableGitHubTeam() *plugin.Table {
+	return &plugin.Table{
+		Name:        "github_team",
+		Description: "GitHub Teams in a given organization. GitHub Teams are groups of organization members that reflect your company or group's structure with cascading access permissions and mentions.",
+		List: &plugin.ListConfig{
+			KeyColumns:        plugin.SingleColumn("organization"),
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			Hydrate:           tableGitHubTeamList,
+		},
+		Get: &plugin.GetConfig{
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "organization", Require: plugin.Optional},
+				{Name: "organization_id", Require: plugin.Optional},
+				{Name: "slug", Require: plugin.Optional},
+				{Name: "id", Require: plugin.Optional},
+			},
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			Hydrate:           tableGitHubTeamGet,
+		},
+		Columns: gitHubOrgTeamColumns(),
+	}
+}
+
+//// LIST FUNCTION
+
+func tableGitHubTeamList(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	client := connect(ctx, d)
+
+	opt := &github.ListOptions{PerPage: 100}
+
+	quals := d.KeyColumnQuals
+	org := quals["organization"].GetStringValue()
+
+	limit := d.QueryContext.Limit
+	if limit != nil {
+		if *limit < int64(opt.PerPage) {
+			opt.PerPage = int(*limit)
+		}
+	}
+
+	type ListPageResponse struct {
+		teams []*github.Team
+		resp  *github.Response
+	}
+
+	listPage := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+		teams, resp, err := client.Teams.ListTeams(ctx, org, opt)
+		return ListPageResponse{
+			teams: teams,
+			resp:  resp,
+		}, err
+	}
+
+	for {
+		listPageResponse, err := plugin.RetryHydrate(ctx, d, h, listPage, &plugin.RetryConfig{ShouldRetryError: shouldRetryError})
+
+		if err != nil {
+			return nil, err
+		}
+
+		listResponse := listPageResponse.(ListPageResponse)
+		teams := listResponse.teams
+		resp := listResponse.resp
+
+		for _, i := range teams {
+			if i != nil {
+				d.StreamListItem(ctx, i)
+			}
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+
+		opt.Page = resp.NextPage
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func tableGitHubTeamGet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	var org string
```

Comment:
Here and below, the code to support both cases is pretty ugly. Suggestions are welcome.